src/core/ext/filters/client_channel/lb_policy_factory.c

```diff
@@ -36,24 +36,50 @@ grpc_lb_addresses* grpc_lb_addresses_create(
   return addresses;
 }
 
+void grpc_lb_address_copy(grpc_lb_address* dest, const grpc_lb_address* src,  
+                          const grpc_lb_user_data_vtable* user_data_vtable) {
+  if (src->user_data != NULL) GPR_ASSERT(user_data_vtable != NULL);
+  memcpy(dest, src, sizeof(grpc_lb_address));
+  if (src->balancer_name != NULL) {
+    dest->balancer_name = gpr_strdup(src->balancer_name);
+  }
+  if (src->user_data != NULL) {
+    dest->user_data = user_data_vtable->copy(src->user_data);
+  }
+}
+
 grpc_lb_addresses* grpc_lb_addresses_copy(const grpc_lb_addresses* addresses) {
   grpc_lb_addresses* new_addresses = grpc_lb_addresses_create(
       addresses->num_addresses, addresses->user_data_vtable);
-  memcpy(new_addresses->addresses, addresses->addresses,
-         sizeof(grpc_lb_address) * addresses->num_addresses);
   for (size_t i = 0; i < addresses->num_addresses; ++i) {
-    if (new_addresses->addresses[i].balancer_name != NULL) {
-      new_addresses->addresses[i].balancer_name =
-          gpr_strdup(new_addresses->addresses[i].balancer_name);
-    }
-    if (new_addresses->addresses[i].user_data != NULL) {
-      new_addresses->addresses[i].user_data = addresses->user_data_vtable->copy(
-          new_addresses->addresses[i].user_data);
-    }
+    grpc_lb_address_copy(&new_addresses->addresses[i], &addresses->addresses[i], 
+                         addresses->user_data_vtable);
   }
   return new_addresses;
 }
 
+grpc_lb_addresses* grpc_lb_addresses_extract_backends(
+    const grpc_lb_addresses* addresses) {
+  // First run: get the number of backends.
+  size_t num_backend_addrs = 0;
+  for (size_t i = 0; i < addresses->num_addresses; ++i) {
+    if (!addresses->addresses[i].is_balancer) ++num_backend_addrs;
+  }
+  if (num_backend_addrs == 0) return NULL;
```

Comment:
In this case, let's return an empty list instead of returning NULL.

I think if we just remove this line, it will do the right thing.